source/src/iWalker.kt

```diff
@@ -1,3 +1,29 @@
 interface iWalker {
+    val curSpeed: Int
     fun walk()
-}
\ No newline at end of file
+    fun stop()
+    fun currentSpeed()
+    fun printAllInfo()
+}
+
+class ZombieWalker(): iWalker {
+    override val curSpeed: Int
+        get() = 1
+
+    override fun walk() {
+        println("BRAINS! BRAINS! BRAINS!")
+    }
+
+    override fun stop() {
+        println("Grrrrr. Ahhhhhh. Uhhhh.")
+    }
+
+    override fun currentSpeed() {
+        println("Zombie speed is $curSpeed km/h")
+    }
+
+    override fun printAllInfo() {
+        this.walk()
```

Comment:
U don't need use this
Its use only when u have more function or var's than one with the same name